server.py

```diff
@@ -23,53 +28,137 @@
+    if (os.path.exists(file_path)):
+        return app.config['HOST'], 200
+    else:
+        return 'File not found', 404
 
+# Helper method for sending a file to another server
 def move_file(request):
     file_uuid = request.args.get('uuid')
     destination = request.args.get('destination')
     destination_with_endpoint = destination + '/write'
-    write_request = util.construct_put_request(destination_with_endpoint, file_uuid)
+    write_request = util.construct_post_request(destination_with_endpoint, file_uuid)
+    metadata = metadata_manager.MetadataManager()
+    metadata.update_file_stored(file_uuid, destination)
+    metadata.close_connection()
     return write_request
 
+# Transfers the file. This API call should not be open to all users.
 @app.route('/transfer', methods=['PUT'])
 def transfer():
     write_request = move_file(request)
     if write_request.status_code == 201:
         os.remove(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid))
+        metadata.delete_file_stored(request.args.get('uuid'), request.args.get('destination'))
+        metadata.close_connection()
     return write_request
 
+# Replicate the file. This API call should not be open to all users.
 @app.route('/replicate', methods=['PUT'])
 def replicate():
     write_request = move_file(request)
     return write_request
 
+# Deletes the file. This API call should not be open to all users.
+@app.route('/delete', methods=['DELETE'])
+def delete():
+    file_uuid = request.args.get('uuid')
+    file_path = UPLOAD_FOLDER + '/' + file_uuid
+    if (os.path.exists(file_path)):
+        os.remove(file_path)
+        metadata = metadata_manager.MetadataManager()
+        metadata.delete_file_stored(file_uuid, app.config['HOST'])
+        metadata.close_connection()
+        return 'Success', 200
+    return 'File not found', 404
+
+# Returns the log.
 @app.route('/logs', methods=['GET'])
 def logs():
     return 'logs'
 
+# Shuts down the server
+@app.route('/shutdown', methods=['GET'])
+def shutdow():
+    func = request.environ.get('werkzeug.server.shutdown')
+    if func is None:
+        raise RuntimeError('Not running with the Werkzeug Server')
+    func()
+    return 'Server is shutting down...', 200
+
+# Entry point for the app
 if __name__ == '__main__':
-    port = None
-    if len(sys.argv) > 1:
-        port = sys.argv[1]
-    if port == None:
-        app.run(host='0.0.0.0', debug=True)
-    else:
-        app.run(host='0.0.0.0', port=int(port))
+    # Default values
+    hostname = '0.0.0.0'
+    port = '5000'
+    server_list = []
+
+    server_list_file = sys.argv[1]
+    # Populate when there are arguments
+    if len(sys.argv) > 2:
+        hostname = sys.argv[2]
+        port = sys.argv[3]
+
+    # Read the file
+    with open(SERVER_LIST_FILE, 'rb') as server_file:
+        server_list = server_file.readlines()
+
+    # Update the metadata
+    metadata = metadata_manager.MetadataManager()
+    metadata.update_server(server_list)
```

Comment:
if we keep running this on the same machine, should we ever clear out the server metadata? if we change the server list file, won't it just keep old records? Maybe i'm missing something